README.md

```diff
@@ -73,12 +73,79 @@ Karafka provides following rake tasks:
 | rake karafka:run     | Runs a single Karafka processing instance |
 | rake karafka:sidekiq | Runs a single Sidekiq worker for Karafka  |
 
+
 ## Sending events from Karafka
 
 To add ability to send events you need add **waterdrop** gem to your Gemfile.
 
 Please follow [WaterDrop README](https://github.com/karafka/waterdrop/blob/master/README.md) for more details on how to install and use it.
 
+## Usage
+
+### Receiving messages
+
+First create application as it was written in **Installation** section above.
+It will generate app folder with controllers and models folder, app.rb file, config folder with sidekiq.yml.example file.
+Also you should have log folder where karafka logs will be written, rakefile.rb file to have ability to run karafka rake tasks.
+
+Example of rakefile.rb:
+
+```ruby
+  ENV['KARAFKA_ENV'] ||= 'development'
+
+  require './app.rb'
+
+  task(:environment) {}
+
+  Karafka::Loader.new.load(Karafka::App.root)
+
+  Dir.glob('lib/tasks/*.rake').each { |r| load r }
+```
+
+Now, to have ability to receive messages you should define controllers in app/controllers folder inherited from Karafka::BaseController. Controller should define *perform* method and has *group* and *topic* names. Names should be unique. You should not define different controllers with the same group or topic names.
+
+You can add any number of *before_enqueue* callbacks. It can be method or block. Write here logic which will not take long time to implement. You have access to params. Based on it you can define which data you want to receive and which not.
```

Comment:
you can add that the before_enqueue acts as rails before_action and that is should perform "lightweight" operations